packages/hardhat-core/src/internal/cli/autocomplete.ts

```diff
@@ -0,0 +1,120 @@
+import { HARDHAT_PARAM_DEFINITIONS } from "../core/params/hardhat-params";
+import type hardhat from "../lib/hardhat-lib";
+
+import { ArgumentsParser } from "./ArgumentsParser";
+
+interface CompletionEnv {
+  line: string;
+  point: number;
+}
+
+interface Suggestion {
+  name: string;
+  description?: string;
+}
+
+export async function complete({
+  line,
+  point,
+}: CompletionEnv): Promise<Suggestion[]> {
+  let hre: typeof hardhat;
+  try {
+    process.env.TS_NODE_TRANSPILE_ONLY = "1";
```

Comment:
I'd move this to `hh`, and that way probably importing the lib would work.

Reply:
I tried this, but it stopped working with ts projects.